server/src/food/handler/FoodItemCategory.handler.ts

```diff
@@ -0,0 +1,66 @@
+import { Request, Response } from "express";
+import { Types } from "mongoose";
+import { DbExtension } from "../../db/interface/db-extension";
+import { FoodItemCategory } from "../index";
+
+export class FoodItemCategoryHandler {
+
+  constructor (private _db: DbExtension) {}
+
+  public getAll = async (req: Request, res: Response): Promise<void> => {
+    try {
+      const result = await this._db.fooditemCategory.getAll();
+      res.status(200).json(result);
+    } catch (error) {
+      res.status(error.statusCode).json(error.forResponse());
+    }
+  }
+
+  public getById = async (req: Request, res: Response): Promise<void> => {
+    try {
+      const id = new Types.ObjectId(req.params.id);
+      const result = await this._db.fooditemCategory.getById(id);
+      res.status(200).json(result);
+    } catch (error) {
+      res.status(error.statusCode).json(error.forResponse());
+    }
+  }
+
+  public save = async (req: Request, res: Response): Promise<void> => {
+    try {
+      const category = new FoodItemCategory().clone(req.body);
+
+      // TODO:  await this._db.fooditem.getById(category.foodItemId);
+
+      const result = await this._db.fooditemCategory.save(category) as Types.ObjectId;
+      res.status(201).json(category.setId(result));
+    } catch (error) {
+      res.status(error.statusCode).json(error.forResponse());
+    }
+  }
+
+  public updateById = async (req: Request, res: Response): Promise<void> => {
+    try {
+      const categoryId = new Types.ObjectId(req.params.id);
+      const foodItemId = req.body.foodItemId;
+      const updatedCategoryName = req.body.name;
+
+      // TODO:  await this._db.fooditem.getById(foodItemId);
```

Comment:
same as above

Reply:
it's not implemented at this time. that's the next step.